### PR TITLE
Fix Challenge Exploit

### PR DESCRIPTION
**Pull request by aumu1031**
Fix players being able to resign while challenged and let their friends claim the seat before the challenger.

### CODE REVIEW

File: classes/village/VillageManager.php [hunk 1]
```diff
@@ -347,25 +347,74 @@ public static function claimSeat(System $system, User $player, string $seat_type
...
+        } else {
+            $message = "No village seat found!";
+        }
+        // if active challenges to your seat, auto win for the challenger
+        if (isset($challenger)) {
+            // verify challenger meets requirements
+            self::checkSeatRequirements($system, $challenger, $player_seat_type);
+            // claim seat for challenger
+            self::claimSeat($system, $challenger, $player_seat_type);
+        }
```
Comment (levimeahan) [inline classes/village/VillageManager.php:394]: Hmm, what if there are multiple challengers?

Comment (levimeahan) [reply classes/village/VillageManager.php:394]: Ah I see the query above pulls the first challenge request 👍

File: classes/village/VillageManager.php [hunk 2]
```diff
@@ -347,25 +347,74 @@ public static function claimSeat(System $system, User $player, string $seat_type
...
      */
     public static function resign(System $system, User $player): string
     {
+        $message = '';
+        $time = time();
+        $player_seat = $player->village_seat;
+        $player_seat_id = $player_seat->seat_id;
+        $player_seat_type = $player_seat->seat_type;
+        // get challenger if set
+        $challenger = null;
```
Comment (levimeahan) [inline classes/village/VillageManager.php:360]:
```suggestion
        $first_challenger = null;
```
to make the intended behavior here more clear

**Review by levimeahan — APPROVED** [review]

File: classes/village/VillageManager.php (hunk 2 above)
Comment (aumu1031) [reply classes/village/VillageManager.php]: > to make the intended behavior here more clear

Done!

**Pull request merged**